rxandroidble/src/main/java/com/polidea/rxandroidble/RxBleDevice.java

```diff
@@ -81,9 +81,38 @@
      * @throws BleGattCallbackTimeoutException emitted when an internal timeout for connection has been reached. The operation will
      *                                         timeout in direct mode (autoConnect = false) after 35 seconds.
      */
+    @Deprecated
     Observable<RxBleConnection> establishConnection(boolean autoConnect);
 
     /**
+     * Establishes connection with a given BLE device. {@link RxBleConnection} is a handle, used to process BLE operations with a connected
+     * device.
+     * <p>
+     * The connection is automatically disconnected (and released) when resulting Observable is unsubscribed.
+     * On the other hand when the connections is interrupted by the device or the system, the Observable will be unsubscribed as well
+     * following BleDisconnectedException or BleGattException emission.
+     * <p>
+     * During the disconnect process the library automatically handles order and requirement of device disconnect and gatt close operations.
+     * <p>
+     * Autoconnect concept may be misleading at first glance. In cases when the BLE device is available and it is advertising constantly you
```

Comment:
This part about `autoConnect` should go to `ConnectionSetup` javadoc as it will be the place to set.